app/javascript/src/component_branch.js

```diff
@@ -83,8 +83,14 @@ class Branch {
   }
 
   destroy() {
+    // find preceding 'or' text to remove as well
+    var $or = this.$node.prev('.branch-or').first();
```

Comment:
'.branch-or' element is design requirement and, I think, not something a Branch object should need to concern itself about (because it also doesn't know of other Branch objects as that is a view controller responsibility).

If view alterations are required, we should put an event broadcaster inside the destroy() method, so that the view controller (set up with a listener) can adjust the view when required.